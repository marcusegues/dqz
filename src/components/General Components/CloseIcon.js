// @flow
import React from 'react';
// $FlowFixMe
import { View, StyleSheet, TouchableWithoutFeedback } from 'react-native';
import MaterialIcons from '@expo/vector-icons/MaterialIcons';

const ownStyles = StyleSheet.create({
  closeIconContainer: {
    alignSelf: 'flex-end',
    paddingRight: 5,
    paddingTop: 5,
  },
});

type CloseIconProps = {
  onPress: Function,
};

export const CloseIcon = ({ onPress }: CloseIconProps) => (
  <View style={ownStyles.closeIconContainer}>
    <TouchableWithoutFeedback onPress={() => onPress()}>
      <MaterialIcons name="close" size={28} color="#DC0018" />
    </TouchableWithoutFeedback>
  </View>
);
