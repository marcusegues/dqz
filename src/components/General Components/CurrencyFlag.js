// @flow
import React from 'react';
// $FlowFixMe
import { Text } from 'react-native';
import type { Currency } from '../../model/currencies';
import { currencyPicker } from '../Modals/CurrencyPickerModal/currencyPickerData';

type CurrencyFlagProps = {
  currency: Currency,
  fontSize?: number,
};

export const CurrencyFlag = ({ currency, fontSize }: CurrencyFlagProps) => {
  const flag = currencyPicker[currency];
  return <Text style={{ fontSize }}>{flag}</Text>;
};

CurrencyFlag.defaultProps = {
  fontSize: 30,
};
