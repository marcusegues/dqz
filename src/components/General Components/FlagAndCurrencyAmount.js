// @flow
import React from 'react';
// $FlowFixMe
import { View, StyleSheet, Text } from 'react-native';
import { CardRowText } from '../QuestionAnswer/cards/subcomponents/CardRowText';
import { scale } from '../../styles/Scaling';
import type { Currency } from '../../model/currencies';

const ownStyles = StyleSheet.create({
  content: {
    flex: 1,
    flexDirection: 'row',
  },
  flag: {
    marginRight: scale(5),
  },
});

type FlagAndCurrencyAmountProps = {
  currency: Currency,
  amount: number,
  flag: Array<string>,
};

export const FlagAndCurrencyAmount = ({
  currency,
  amount,
  flag,
}: FlagAndCurrencyAmountProps) => (
  <View style={ownStyles.content}>
    <Text style={ownStyles.flag}>{flag}</Text>
    <CardRowText text={`${currency} ${amount}`} />
  </View>
);
