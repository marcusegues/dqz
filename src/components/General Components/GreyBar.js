// @flow
import React from 'react';
// $FlowFixMe
import { View } from 'react-native';
import { GREY_BAR } from '../../styles/colors';

export const GreyBar = () => (
  <View
    style={{
      width: 8,
      height: 4,
      alignSelf: 'flex-end',
      right: 24,
      backgroundColor: GREY_BAR,
    }}
  />
);
