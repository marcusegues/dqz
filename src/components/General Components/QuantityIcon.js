// @flow
import React from 'react';
// $FlowFixMe
import { View, Text, StyleSheet } from 'react-native';

const ownStyles = StyleSheet.create({
  quantityContainer: {
    paddingHorizontal: 8,
    paddingVertical: 2,
    backgroundColor: '#E0E0E1',
    borderRadius: 3,
  },
  quantityText: {
    color: 'black',
    fontSize: 14,
    fontFamily: 'roboto_regular',
  },
});

type QuantityIconProps = {
  quantity: number,
  unit: string,
};

const QuantityIcon = ({ quantity, unit }: QuantityIconProps) => (
  <View style={ownStyles.quantityContainer}>
    <Text style={ownStyles.quantityText}>{`${quantity} ${unit}`}</Text>
  </View>
);

export default QuantityIcon;
