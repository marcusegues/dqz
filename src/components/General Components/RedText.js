// @flow
import React from 'react';
// $FlowFixMe
import { Text } from 'react-native';
import { moderateScale } from '../../styles/Scaling';
import { MAIN_RED } from '../../styles/colors';
import { size, type } from '../../styles/fonts';

const ownStyles = {
  text: {
    color: MAIN_RED,
    fontFamily: type.medium,
    fontSize: moderateScale(size.medium),
  },
};

type RedTextProps = {
  text: string,
  style?: {},
};

export const RedText = ({ text, style }: RedTextProps) => (
  <Text style={[ownStyles.text, { ...style }]}>{text}</Text>
);

RedText.defaultProps = {
  style: {},
};
