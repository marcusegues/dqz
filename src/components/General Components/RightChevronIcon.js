import React from 'react';
import { View } from 'react-native';
import Entypo from '@expo/vector-icons/Entypo';
import { moderateScale } from '../../styles/Scaling';

const RightChevronIcon = () => (
  <View style={{ alignSelf: 'center' }}>
    <Entypo name="chevron-right" size={moderateScale(25)} color="#24253D" />
  </View>
);

export default RightChevronIcon;
