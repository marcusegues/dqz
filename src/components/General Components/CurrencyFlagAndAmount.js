// @flow
import React from 'react';
// $FlowFixMe
import { View, StyleSheet } from 'react-native';
import { CardRowText } from '../QuestionAnswer/Cards/subcomponents/CardRowText';
import { scale } from '../../styles/Scaling';
import type { Currency } from '../../model/currencies';
import { CurrencyFlag } from './CurrencyFlag';

const ownStyles = StyleSheet.create({
  content: {
    flex: 1,
    flexDirection: 'row',
    alignItems: 'center',
  },
});

type CurrencyFlagAndAmountProps = {
  currency: Currency,
  amount: number,
};

export const CurrencyFlagAndAmount = ({
  currency,
  amount,
}: CurrencyFlagAndAmountProps) => (
  <View style={ownStyles.content}>
    <View style={{ marginRight: scale(5) }}>
      <CurrencyFlag currency={currency} />
    </View>
    <CardRowText text={`${currency} ${amount}`} />
  </View>
);
