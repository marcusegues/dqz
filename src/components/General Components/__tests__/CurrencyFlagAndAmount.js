// @flow
import React from 'react';
import renderer from 'react-test-renderer';
import { shallow } from 'enzyme';
import { CurrencyFlagAndAmount } from '../../General Components/CurrencyFlagAndAmount';
import { CurrencyFlag } from '../CurrencyFlag';
import { CardRowText } from '../../QuestionAnswer/Cards/subcomponents/CardRowText';

describe('CurrencyFlagAndAmount TestSuite', () => {
  test('CurrencyFlagAndAmount component renders correctly', () => {
    const tree = renderer
      .create(<CurrencyFlagAndAmount currency="EUR" amount={22} />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
  test('CurrencyFlagAndAmount shallow rendered component renders correctly', () => {
    const wrapper = shallow(
      <CurrencyFlagAndAmount currency="EUR" amount={22.444555} />
    ).dive();
    expect(wrapper.find(CurrencyFlag).prop('currency')).toEqual('EUR');
    expect(wrapper.find(CardRowText).prop('text')).toEqual('EUR 22.44');
  });
});
