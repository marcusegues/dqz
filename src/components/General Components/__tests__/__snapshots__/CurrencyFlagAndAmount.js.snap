// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CurrencyFlagAndAmount TestSuite CurrencyFlagAndAmount component renders correctly 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "flex": 1,
      "flexDirection": "row",
    }
  }
>
  <View
    style={
      Object {
        "marginRight": 10.714285714285714,
      }
    }
  >
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Object {
          "fontSize": 30,
        }
      }
    >
      ðŸ‡ªðŸ‡º
    </Text>
  </View>
  <View
    style={Object {}}
  >
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Object {
          "color": "#24253D",
          "fontFamily": "roboto_medium",
          "fontSize": 22,
        }
      }
    >
      EUR 22.00
    </Text>
  </View>
</View>
`;
